apis/metal3.io/v1alpha1/preprovisioningimage_types.go

```diff
@@ -60,15 +60,15 @@ const (
 	ConditionImageError ImageStatusConditionType = "Error"
 )
 
-// PreprovisioningImageStatus defines the observed state of PreprovisioningImage
+// PreprovisioningImageStatus defines the observed state of PreprovisioningImage.
 type PreprovisioningImageStatus struct {
 	// imageUrl is the URL from which the built image can be downloaded.
-	ImageUrl string `json:"imageUrl,omitempty"`
+	ImageUrl string `json:"imageUrl,omitempty"` //nolint:revive,stylecheck
 
 	// kernelUrl is the URL from which the kernel of the image can be downloaded.
 	// Only makes sense for initrd images.
 	// +optional
-	KernelUrl string `json:"kernelUrl,omitempty"`
+	KernelUrl string `json:"kernelUrl,omitempty"` //nolint:revive,stylecheck
```

Comment:
ideally this type name should have been `KernelURL` as asked by the linters, I was not sure if I should do it here since this will cause an API change. I am up for suggestions.

Reply:
same as above